simulator/infra/instrcache/instr_cache.h

```diff
@@ -111,12 +111,11 @@ class LRUCache
         };
 
         std::unordered_map<Key, size_t> data{};
-        std::list<Key> lru_list{};
         std::list<size_t> free_list{};
         std::unique_ptr<void, Deleter> arena = nullptr;
         Value* storage = nullptr;
-        std::unordered_map<Key, typename std::list<Key>::const_iterator> lru_hash{};
+        CacheReplacementInterface* lru_module;
```

Comment:
Please use std::unique_ptr here for simpler memory management (no need to call "delete" explicitly)